src/main/kotlin/org/rust/lang/core/types/RsPsiTypeImplUtil.kt

```diff
@@ -20,9 +20,10 @@ object RsPsiTypeImplUtil {
         val typeReference = psi.typeReference
         if (typeReference != null) return typeReference.type
         return when (psi.owner) {
-            is RsTypeAliasOwner.Free -> TyUnknown
-            is RsTypeAliasOwner.Trait -> TyTypeParameter.associated(psi)
-            is RsTypeAliasOwner.Impl -> TyUnknown
+            is RsAbstractableOwner.Free -> TyUnknown
+            is RsAbstractableOwner.Trait -> TyTypeParameter.associated(psi)
+            is RsAbstractableOwner.Impl -> TyUnknown
+            is RsAbstractableOwner.Foreign -> TyUnknown
```

Comment:
This `is RsAbstractableOwner.Foreign` is new because of the unification.

Reply:
I think it's acceptable price for this unification